04 auth - Done

Shpakk2 opened the pull request:
Перевірте, будь ласка

Shpakk2 committed:
1

Shpakk2 committed:
додав фільтрацію favorite)

Shpakk2 committed:
додав можливість зміни subscription

Eastvarga commented on controllers/auth.js:
```diff
@@ -0,0 +1,82 @@
+        email,
+        subscription,
+    })
+}
```
не зкщзумів і де апдейт підписки?

Eastvarga commented on controllers/contacts.js:
```diff
@@ -5,8 +5,18 @@ const {HttpError, ctrlWrapper} = require('../helpers')
+            skip,
+            limit: +limit,
+        });
+        console.log(favoriteContacts)
```
ніяких консолей у працюючому коді

Eastvarga commented on controllers/contacts.js:
```diff
@@ -5,8 +5,18 @@ const {HttpError, ctrlWrapper} = require('../helpers')
+        console.log(favoriteContacts)
+        return res.json(favoriteContacts);
+    }
+    const result = await Contact.find({owner}, "-createdAt, -updatedAt", {skip, limit}).populate("owner", "email")
```
Щось дивне?
може краще зробити опції для запиту, та формувати опції залежно від favorite, а потім з опціями робити запит до БД,  ніж обрати два варіанти запиту? І це ще питаня чи однакова форма відповіді у цих запитів

Eastvarga commented on middlewares/authenticate.js:
```diff
@@ -0,0 +1,28 @@
+const authenticate = async (req, res, next) => {
+    const { authorization = "" } = req.headers;
+    const  [bearer, token] = authorization.split(" ");
+    if (bearer !== "Bearer") {
```
а якщо токен буде не визначений? чому його не перевіряєте?

Eastvarga commented on models/user.js:
```diff
@@ -0,0 +1,60 @@
+
+userSchema.post('save', mongooseHandleError)
+
+const registerSchema = Joi.object({
+    password: Joi.string().min(6).required(),
+    email: Joi.string().pattern(emailRegexp).message("валідність вашого email викликає підозри").required(),    
+    subscription: Joi.string().required(),
+}) 
+
+const loginSchema = Joi.object({
+    password: Joi.string().min(6).required(),
+    email: Joi.string().pattern(emailRegexp).required(),    
+}) 
+
+const updateSubscriptionSchema = Joi.object({
+  subscription: Joi.string().valid('starter', 'pro', 'business').required(), 
+})
```
схема валідації та схема моделі БД різні речі не змішуйте їх - по різним файлам Валідація \ БазаДанних

Eastvarga reviewed (changes requested):
Є зауваження

Shpakk2 committed:
change

- прибрав консоль лог
-  зробив апдейт підписки
- спростив функцію getAll у controllers/contacts.js
